src/characters/cbd-recipient.ts

```diff
@@ -62,11 +66,32 @@ export class ThresholdDecrypter {
     );
   }
 
+  private async makeAcp(
+    provider: ethers.providers.Web3Provider,
+    conditionExpr: ConditionExpression,
+    ciphertext: Ciphertext
+  ) {
+    const dkgRitual = await DkgClient.getExistingRitual(
+      provider,
+      this.ritualId
+    );
+    const authData = new AuthenticatedData(
+      dkgRitual.dkgPublicKey,
+      conditionExpr.toWASMConditions()
+    );
+
+    const headerHash = keccak256(ciphertext.header.toBytes());
+    const authorization = await provider.getSigner().signMessage(headerHash);
+
+    return new AccessControlPolicy(authData, toBytes(authorization));
+  }
+
   // Retrieve decryption shares
   public async retrieve(
     provider: ethers.providers.Web3Provider,
     conditionExpr: ConditionExpression,
-    ciphertext: Ciphertext
+    ciphertext: Ciphertext,
+    acp: AccessControlPolicy
```

Comment:
@piotr-roslaniec Should this just be `ThreshodlMessageKit`? It has the ciphertext and the acp in it...?